packages/sku/src/services/vite/loadable/loadable.tsx

```diff
@@ -17,13 +17,23 @@ export type PreloadableComponent<T extends ComponentType<any>> = T & {
 };
 
 export function loadable<T extends ComponentType<any>>(
-  factory: () => Promise<{ default: T }>,
+  factory: () => Promise<{ default: T } & Record<string, T>>,
   options?: {
     fallback?: NonNullable<ReactNode> | null;
+    resolveComponent?: (module: { default: T } & Record<string, T>) => T;
   },
   moduleId: ModuleId = '', // Gets set via the plugin
 ): PreloadableComponent<T> {
-  const ReactLazyComponent = lazy(factory);
+  const getDefaultExport = (module: { default: T } & Record<string, T>) =>
```

Comment:
Feel like this could have a better name that's not _only_ related to default exports:
```suggestion
  const getResolvedOrDefaultExport = (module: { default: T } & Record<string, T>) =>
```